internal/services/zero_trust_access_application/schema.go

```diff
@@ -1326,52 +1426,86 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 					"default_relay_state": schema.StringAttribute{
 						Description: "The URL that the user will be redirected to after a successful login for IDP initiated logins.",
 						Optional:    true,
+						Validators: []validator.String{
+							customvalidator.RequiresOtherStringAttributeToBeOneOf(path.MatchRoot("saas_app").AtName("auth_type"), "saml"),
+						},
 					},
 					"idp_entity_id": schema.StringAttribute{
 						Description: "The unique identifier for your SaaS application.",
 						Computed:    true,
 						Optional:    true,
+						Validators: []validator.String{
+							customvalidator.RequiresOtherStringAttributeToBeOneOf(path.MatchRoot("saas_app").AtName("auth_type"), "saml"),
+						},
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
 					},
 					"name_id_format": schema.StringAttribute{
 						Description: "The format of the name identifier sent to the SaaS application.\nAvailable values: \"id\", \"email\".",
 						Optional:    true,
+						Computed:    true,
 						Validators: []validator.String{
 							stringvalidator.OneOfCaseInsensitive("id", "email"),
+							customvalidator.RequiresOtherStringAttributeToBeOneOf(path.MatchRoot("saas_app").AtName("auth_type"), "saml"),
+						},
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
 						},
 					},
 					"name_id_transform_jsonata": schema.StringAttribute{
 						Description: "A [JSONata](https://jsonata.org/) expression that transforms an application's user identities into a NameID value for its SAML assertion. This expression should evaluate to a singular string. The output of this expression can override the `name_id_format` setting.",
 						Optional:    true,
+						Validators: []validator.String{
+							customvalidator.RequiresOtherStringAttributeToBeOneOf(path.MatchRoot("saas_app").AtName("auth_type"), "saml"),
```

Comment:
Is name id transform only available on saml?

Reply:
correct, I never got around to adding it for oidc (and we'd name it something else probably)